packages/docusaurus-plugin-client-redirects/src/__tests__/writeRedirectFiles.test.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import fs from 'fs-extra';
+import path from 'path';
+
+import writeRedirectFiles, {
+  toRedirectFilesMetadata,
+} from '../writeRedirectFiles';
+
+describe('toRedirectFilesMetadata', () => {
+  test('should create appropriate metadatas', async () => {
+    const pluginContext = {
+      outDir: '/tmp/someFixedOutDir',
+      baseUrl: 'https://docusaurus.io',
+    };
+
+    const redirectFiles = toRedirectFilesMetadata(
+      [
+        {fromRoutePath: '/abc.html', toRoutePath: '/abc'},
+        {fromRoutePath: '/def', toRoutePath: '/def.html'},
+        {fromRoutePath: '/xyz', toRoutePath: '/'},
+      ],
+      pluginContext,
+    );
+
+    expect(redirectFiles.map((f) => f.fileAbsolutePath)).toEqual([
+      path.join(pluginContext.outDir, '/abc.html/index.html'),
```

Comment:
@lex111 @yangshun , this is some part I'm not totally sure of.

If we need to redirect from path `/abc.html`, should we write `/abc.html` to the disk or `/abc.html/index.html` ? Is there a method that already converts a route path to a fs file path somewhere?